src/main/java/org/semux/core/TransactionExecutor.java

```diff
@@ -60,8 +72,9 @@ public static boolean validateDelegateName(byte[] data) {
      * 
      * @param config
      */
-    public TransactionExecutor(Config config) {
+    public TransactionExecutor(Config config, Blockchain blockchain) {
```

Comment:
This is a very strong dependency. What's the rational behind it?

Reply:
Replaced in favor of using BlockStore interface.  Thanks!